src/js/containers/agency/v2/accountSpending/TableContainer.jsx

```diff
@@ -0,0 +1,200 @@
+    const [sort, setSort] = useState('obligatedAmount');
+    const [order, setOrder] = useState('desc');
+    const updateSort = (field, direction) => {
+        setSort(field);
+        setOrder(direction);
+    };
+    const [results, setResults] = useState([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(false);
+    // TODO - use totalObligation to calculate '% of Total Obligations' column value
+    const totalObligation = useSelector((state) => state.agencyV2.budgetaryResources._agencyTotalObligated);
+
+    const parseAccount = (data) => {
+        if (totalObligation) {
+            // add total obligation to each object and it's children objects
+            const dataAndTotalObligation = data.map((d) => {
+                let dataChildrenAndTotalObligation = [];
+                if (d.children && d.children.length > 0) {
+                    dataChildrenAndTotalObligation = d.children.map((child) => ({
+                        ...child,
+                        total_obligated_amount: totalObligation
+                    }));
+                }
+
+                if (dataChildrenAndTotalObligation.length > 0) {
+                    return {
+                        ...d,
+                        children: dataChildrenAndTotalObligation,
+                        total_obligated_amount: totalObligation
+                    };
+                }
+
+                return {
+                    ...d,
+                    total_obligated_amount: totalObligation
+                };
+            });
+
+            // parse row and row's children
+            const parsedData = dataAndTotalObligation.map((item) => {
+                const accountSpendingRow = Object.create(BaseAccountSpendingRow);
+                accountSpendingRow.populate(item);
+
+                let rowChildren = [];
+                if (item.children && item.children.length > 0) {
+                    rowChildren = item.children.map((childItem) => {
+                        const accountChildSpendingRow = Object.create(BaseAccountSpendingRow);
+                        accountChildSpendingRow.populate(childItem);
+                        return accountChildSpendingRow;
+                    });
+                }
+
+                if (rowChildren && rowChildren.length > 0) {
+                    Object.defineProperty(accountSpendingRow, "children", {
+                        value: rowChildren
+                    });
+                }
+
+                return accountSpendingRow;
+            });
+            setResults(parsedData);
+        }
+    };
+
+    useEffect(() => {
+        setLoading(true);
+        // Reset to the first page
+        changePage(1);
+        const params = {
+            fiscal_year: props.fy,
+            limit: pageSize,
+            sort: accountFields[sort],
+            order
+        };
+        const request = fetchSpendingByCategory(props.agencyId, props.type, params);
```

Comment:
We should migrate this to it's own function with `useCallback()` since we are doing the same thing in two effects with the only difference `setTotalItems(res.data.page_metadata.total);` and we can just do that both times or add some logic to see it it has been populated and then call it.

Reply:
I created a function using `useCallback()`, but not sure it's doing anything because we're not passing anything in its dependency array. Let me know, if you have any idea how it should work.

Reply:
Looking good. Thank you. We cannot pass parameters to `useCallback` that I am aware of. So we would need to move 
```
        setLoading(true);
        // Make a request with the new page number
        const params = {
            fiscal_year: props.fy,
            limit: pageSize,
            page: currentPage,
            sort: accountFields[sort],
            order
        };
```
into `fetchSpendingByCategoryCallback ` as well and then it should work.

Reply:
Done!